File: src/main/java/org/shark/example/service/mysql/FindJsonInnerJoinResultService.java
package org.shark.example.service.mysql;

import com.google.gson.Gson;
import lombok.RequiredArgsConstructor;
import org.shark.example.controller.mysql.pojo.JsonInnerJoinDto;
import org.shark.example.dao.repository.JsonExampleRepository;
import org.shark.example.dao.repository.pojo.JsonInnerJoinDo;
import org.shark.example.service.mysql.pojo.CreateJsonExampleDto;
import org.springframework.stereotype.Service;

import java.util.ArrayList;
import java.util.List;

@RequiredArgsConstructor
@Service
public class FindJsonInnerJoinResultService {
    private final JsonExampleRepository jsonExampleRepository;

    public List<JsonInnerJoinDto> start() {
        Gson gson = new Gson();
        List<JsonInnerJoinDo> jsonInnerJoinDoList = jsonExampleRepository.findAllJoin();
        List<JsonInnerJoinDto> list = new ArrayList<>();
        for(JsonInnerJoinDo jsonInnerJoinDo: jsonInnerJoinDoList) {
            JsonInnerJoinDto jsonInnerJoinDto = new JsonInnerJoinDto();
            jsonInnerJoinDto.setId(jsonInnerJoinDo.getId());
            jsonInnerJoinDto.setDescription(jsonInnerJoinDo.getDescription());
            jsonInnerJoinDto.setCreateJsonExampleDto(gson.fromJson(jsonInnerJoinDo.getJsonData(), CreateJsonExampleDto.class));
            list.add(jsonInnerJoinDto);
        }
        return list ;
    }
}

File: src/main/kotlin/org/shark/example/service/mysql/FindJsonInnerJoinResultService.kt
package org.shark.example.service.mysql

import com.google.gson.Gson
import org.shark.example.controller.mysql.pojo.JsonInnerJoinDto
import org.shark.example.dao.repository.JsonExampleRepository
import org.shark.example.service.mysql.pojo.CreateJsonExampleDto
import org.springframework.stereotype.Service

@Service
class FindJsonInnerJoinResultService(
    private val jsonExampleRepository: JsonExampleRepository
) {
    fun start(): List<JsonInnerJoinDto> {
        val gson = Gson()
        val jsonInnerJoinDoList = jsonExampleRepository.findAllJoin()
        val list = ArrayList<JsonInnerJoinDto>()
        for (jsonInnerJoinDo in jsonInnerJoinDoList) {
            val jsonInnerJoinDto = JsonInnerJoinDto()
            jsonInnerJoinDto.id = jsonInnerJoinDo.id
            jsonInnerJoinDto.description = jsonInnerJoinDo.description
            jsonInnerJoinDto.createJsonExampleDto = gson.fromJson(jsonInnerJoinDo.jsonData, CreateJsonExampleDto::class.java)
            list.add(jsonInnerJoinDto)
        }
        return list
    }
}
